<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>pwmEditForm</class>
 <widget class="QWidget" name="pwmEditForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>832</width>
    <height>674</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetDefaultConstraint</enum>
   </property>
   <property name="horizontalSpacing">
    <number>1</number>
   </property>
   <item row="2" column="2" colspan="2">
    <widget class="QLineEdit" name="nameLineEdit">
     <property name="minimumSize">
      <size>
       <width>250</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>350</width>
       <height>16777215</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_18">
     <property name="text">
      <string>Arduino pin:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_19">
     <property name="text">
      <string>ID</string>
     </property>
    </widget>
   </item>
   <item row="6" column="5">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0" colspan="6">
    <widget class="QLabel" name="label_8">
     <property name="font">
      <font>
       <pointsize>11</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: white;
background: rgb(0, 151, 157);
padding: 3px
</string>
     </property>
     <property name="text">
      <string>Edit Instrument</string>
     </property>
    </widget>
   </item>
   <item row="9" column="2">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>26</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="2">
    <widget class="QComboBox" name="PIN_comboBox">
     <property name="maximumSize">
      <size>
       <width>50</width>
       <height>16777215</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="6" column="2">
    <widget class="QLabel" name="drefInfoLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>350</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="text">
      <string>Name</string>
     </property>
    </widget>
   </item>
   <item row="6" column="4">
    <widget class="QLabel" name="drefValueLabel">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="3" column="2" colspan="2">
    <widget class="QLineEdit" name="IDlineEdit">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="minimumSize">
      <size>
       <width>250</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>350</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>nameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>pwmEditForm</receiver>
   <slot>updateXMLdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>368</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PIN_comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>pwmEditForm</receiver>
   <slot>updateXMLdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>487</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PIN_comboBox</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>pwmEditForm</receiver>
   <slot>updatePin()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addCommand()</slot>
  <slot>rmCommand()</slot>
  <slot>addDataref()</slot>
  <slot>rmDataref()</slot>
  <slot>editXPCommand()</slot>
  <slot>updateXMLdata()</slot>
  <slot>testCommands()</slot>
  <slot>updatePin()</slot>
  <slot>editXPDataref()</slot>
  <slot>testDatarefs()</slot>
  <slot>activateSave()</slot>
 </slots>
</ui>
